sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-jedis/src/main/java/com/azure/messaging/eventhubs/checkpointstore/jedis/JedisRedisCheckpointStore.java

```diff
@@ -0,0 +1,152 @@
+import reactor.core.Exceptions;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import redis.clients.jedis.JedisPool;
+import redis.clients.jedis.Jedis;
+
+/**
+ * Implementation of {@link CheckpointStore} that uses Azure Redis Cache, specifically Jedis.
+ */
+public class JedisRedisCheckpointStore implements CheckpointStore {
+    private static final ClientLogger LOGGER = new ClientLogger(JedisRedisCheckpointStore.class);
+    private static final JsonSerializer DEFAULT_SERIALIZER = JsonSerializerProviders.createInstance(true);
+    static final String CHECKPOINT = "checkpoint";
+    static final String  PARTITION_OWNERSHIP = "partitionOwnership";
+    private final JedisPool jedisPool;
+    JedisRedisCheckpointStore(JedisPool jedisPool) {
+        this.jedisPool = jedisPool;
+    }
+
+    /**
+     * This method returns the list of partitions that were owned successfully.
+     *
+     * @param requestedPartitionOwnerships List of partition ownerships from the current instance
+     * @return Null
+     */
+    @Override
+    public Flux<PartitionOwnership> claimOwnership(List<PartitionOwnership> requestedPartitionOwnerships) {
+        return null;
+    }
+
+    /**
+     * This method returns the list of checkpoints from the underlying data store, and if no checkpoints are available, then it returns empty results.
+     *
+     * @param fullyQualifiedNamespace The fully qualified namespace of the current instance of Event Hub
+     * @param eventHubName The Event Hub name from which checkpoint information is acquired
+     * @param consumerGroup The consumer group name associated with the checkpoint
+     * @return Null
+     */
+    @Override
+    public Flux<Checkpoint> listCheckpoints(String fullyQualifiedNamespace, String eventHubName, String consumerGroup) {
+        String prefix = prefixBuilder(fullyQualifiedNamespace, eventHubName, consumerGroup);
+        try (Jedis jedis = jedisPool.getResource()) {
+            ArrayList<Checkpoint> listStoredCheckpoints = new ArrayList<>();
+            Set<String> members = jedis.smembers(prefix);
+            if (members.isEmpty()) {
+                jedisPool.returnResource(jedis);
+                return Flux.fromIterable(listStoredCheckpoints);
+            }
+            for (String member : members) {
+                //get the associated JSON representation for each for the members
+                List<String> checkpointJsonList = jedis.hmget(member, CHECKPOINT);
+                jedisPool.returnResource(jedis);
+                if (checkpointJsonList == null) {
+                    return Flux.error(new IllegalStateException("No checkpoints persist in Redis for the given parameters."));
+                }
+                else {
+                    String checkpointJson = checkpointJsonList.get(0);
```

Comment:
What if the list is empty? this will return an index out of bounds exception.... same with usage below.